lib/tasks/check_license.rake

```diff
@@ -113,9 +113,9 @@ def license_report
   if ! report[:missing].empty?
     raise "\nMissing license:\n#{report[:missing].join("\n")}"
   end
-  puts "\nSkipped files:\n#{report[:skipped].join("\n")}" if verbose
-  puts "\nCopyright found in these files:\n#{report[:seen].join("\n")}" if verbose
-  puts "\nCopyright detected as not needed in these files:\n#{report[:unneeded].join("\n")}" if verbose
+  puts "\nSkipped files:\n#{report[:skipped].join("\n")}" if verbose == true
+  puts "\nCopyright found in these files:\n#{report[:seen].join("\n")}" if verbose == true
+  puts "\nCopyright detected as not needed in these files:\n#{report[:unneeded].join("\n")}" if verbose == true
   puts "\nAll files have proper license reference." if verbose
```

Comment:
Why sometimes `verbose == true` ?

Reply:
With `--silent` it is `false`, with `--verbose` it is `true`.

But if you do NOT specify anything it's an `Object` (not `nil` or something like that) and it behaves like `true`.

So if you want to print something only with `--verbose` you have to explicitly compare it with `true`.

Reply:
It's because [`verbose`](https://github.com/ruby/rake/blob/5c87c462b64aad674ebb92b1f5b0ff2c911406cd/lib/rake/file_utils_ext.rb#L43-L54) is a tri-state attribute.
- `rake -v/--verbose` sets it to `true`
- `rake -q/--quiet` sets it to `false`
- by default it has a truthy value of `Object.new` which is interpreted as "in-between"

Reply:
Fuzzy logic ;)